File: LexiconTestTasks/src/Character.cpp
#include <iostream>
#include <random>
#include "Character.h"

Character::Character(const std::wstring name)
{
	this->name = name;
	randomizeCharacteristics();
}

void Character::print()
{
	std::wcout << L"Namn: " << name << std::endl
		<< L" Hälsa: " << health << std::endl
		<< L" Styrka: " << strength << std::endl
		<< L" Tur: " << luck << std::endl;
}

uint32_t Character::genRand(uint32_t min, uint32_t max)
{
	return (std::rand() % (max - min + 1) + min);
}

void Character::randomizeCharacteristics()
{
	health = genRand(10, 30);
	strength = genRand(5, 15);
	luck = genRand(2, 8);
}

File: LexiconTestTasksCsharp/Character.cs
using System;

namespace LexiconTestTasksCsharp
{
    class Character
    {
        public Character(string name)
        {
            Name = name;
            InitializeStatistics();
        }

        private void InitializeStatistics()
        {
            Health = random.Next(20, 30);
            Strength = random.Next(10, 15);
            Luck = random.Next(0, 5);
        }

        public void PrintCharacter()
        {
            Console.WriteLine(
                $"[{Name}]\n" +
                $" Health: {Health}\n" +
                $" Strength: {Strength}\n" +
                $" Luck: {Luck}"
                );
        }

        public string Name { get; private set; }
        public int Health { get; private set; }
        public int Strength { get; private set; }
        public int Luck { get; private set; }

        private static Random random = new Random();
    }
}
